client/state/query-client.ts

```diff
@@ -26,6 +50,25 @@ export async function createQueryClient( userId: number | undefined ): Promise<
 			queryClient,
 			persistor,
 			maxAge: MAX_AGE,
+			dehydrateOptions: {
+				shouldDehydrateQuery: ( query ) => {
+					if ( query.state.status !== 'success' ) {
```

Comment:
Should this be customizable? Perhaps in our use case, we're totally fine to persist unsuccessful queries?

Reply:
`success` in this case means that the query didn't fail to arrive and is not an error code. Not really that the intent was achieved!

Reply:
Are we keeping in mind that many of our endpoints have the HTTP envelope and they'll all land as success (OK 200), and that might be tricking the system that they succeeded, while they could actually be containing an error inside?

Reply:
Yeah, that's why it's possible to determine programmatically if we want to store it or not :D

Reply:
> Are we keeping in mind that many of our endpoints have the HTTP envelope and they'll all land as success (OK 200)

However, the `wpcom.req` code will unwrap the envelope, see the error, and will return a rejected promise. The React Query code doesn't look at HTTP status codes. It merely calls the `queryFn` function, which is completely generic, and looks at the result.

The HTTP envelopes are invisible to the `wpcom` library user, they are an implementation detail of the transport layer, where the errors are not encoded as HTTP status codes, but are in the JSON response instead.